appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -62,26 +77,104 @@ public String getName() {
 
     @ReactMethod
     public void read(String documentId, String partition, ReadableMap readOptionsMap, final Promise promise) {
-        ReadOptions readOptions = getReadOptions(readOptionsMap);
+        ReadOptions readOptions = AppCenterReactNativeDataUtils.getReadOptions(readOptionsMap);
         Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer<JsonElement>("Read failed", promise));
     }
 
+    @ReactMethod
+    public void list(String partition, final Promise promise) {
+        Data.list(JsonElement.class, partition).thenAccept(new AppCenterConsumer<PaginatedDocuments<JsonElement>>() {
+
+            @Override
+            public void accept(PaginatedDocuments<JsonElement> documentWrappers) {
+                String paginatedDocumentsId = UUID.randomUUID().toString();
+                mPaginatedDocuments.put(paginatedDocumentsId, documentWrappers);
+                WritableMap paginatedDocumentsMap = new WritableNativeMap();
+                WritableMap currentPageMap = new WritableNativeMap();
+                WritableArray itemsArray = new WritableNativeArray();
+                Page<JsonElement> currentPage = documentWrappers.getCurrentPage();
+                if (currentPage.getError() != null) {
+                    DataException dataException = currentPage.getError();
+                    promise.reject("Failed list", dataException.getMessage(), dataException);
+                    return;
+                }
+                List<DocumentWrapper<JsonElement>> documents = currentPage.getItems();
+
+                /* Add documents to WritableArray */
+                for (DocumentWrapper<JsonElement> document : documents) {
```

Comment:
This crashes with null document

Reply:
Fixed, please refer to commit: [358eb69c116b35bf9c4cde8acf46874965122bfa](https://github.com/microsoft/appcenter-sdk-react-native/pull/636/commits/358eb69c116b35bf9c4cde8acf46874965122bfa)